test(latte): add 5000 tables test cases using latte

vponomaryov opened the pull request:
QA-task: https://github.com/scylladb/qa-tasks/issues/1882

This is test scenario that uses common longevity test and is able to create any amount of keyspaces and tables.

Depends on the following:
- https://github.com/scylladb/scylla-cluster-tests/pull/12906
- https://github.com/scylladb/scylla-cluster-tests/pull/12954
- Latest latte -> https://github.com/scylladb/latte/releases/tag/0.43.0-scylladb

Features of the rune script:
- Any combinations of keyspaces, tables and amount of rows per table
- Tablets/vnodes
- Index creation ON/OFF
- Flexible control of the workload per group of tables
  - Table group size control
  - Table count to shift control
  - Time interval to make shifts control
  - It is directly related to the cache hit/miss control - we will have a relatively stable load with a relatively stable (controlled) cache hit/miss ratio using all of the tables smoothly switching/shifting among them.
  - Data distribution: Uniform / Gauss
    - Uniform useful for data population
    - Gauss/Normal useful for testing read/mixed workloads
    - Custom size for the data
  - Data validation ON/OFF + retries ON/OFF
- Specifics:
  - The thousands of tables get created by the latte benchmarking tool using its “measurement” approach (`latte run …`) where we see the time frames with “good” and “slow/postponed” processing of multiple “create table” queries.
 
### Testing
<!-- Add links to Argus/Jenkins of test test done with this PR -->
<!-- This would help the reviewer to cross check what was tested, and and review the results as needed -->
- [ ]

### PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

vponomaryov commented:
Following Scylla bugs were found using this new approach:
- https://github.com/scylladb/scylladb/issues/27551
- https://github.com/scylladb/scylladb/issues/27566
- https://github.com/scylladb/scylladb/issues/27620
- https://github.com/scylladb/scylladb/issues/27621

vponomaryov commented on test-cases/scale/longevity-5000-tables-latte.yaml:
```diff
@@ -0,0 +1,77 @@
+    -P compaction_strategy="\"IncrementalCompactionStrategy\""
+    -P sstable_compression="\"LZ4Compressor\""
+    data_dir/latte/schema_scale.rn
+prepare_verify_cmd:
```
This phase discloses bugs:
- https://github.com/scylladb/scylladb/issues/27566
- https://github.com/scylladb/scylladb/issues/27620
- https://github.com/scylladb/scylladb/issues/27621

vponomaryov commented on test-cases/scale/longevity-5000-tables-latte.yaml:
```diff
@@ -0,0 +1,77 @@
+    -P table_shift_interval_s=0
+    data_dir/latte/schema_scale.rn
+# TODO: make mixed workload take 36000s (10h) after debugging
+stress_cmd:
```
`vnodes` setup reached this step here: [scylla-staging/valerii/vp-longevity-5000-tables-latte#4](https://argus.scylladb.com/tests/scylla-cluster-tests/3f8e2aac-22f2-42cb-a507-8bc55a84a605)

fruch commented on data_dir/latte/schema_scale.rn:
```diff
@@ -0,0 +1,277 @@
+const TABLE_GROUP_SIZE = latte::param!("table_group_size", 10);
+const TABLE_SHIFT_COUNT = latte::param!("table_shift_count", 1);
+// NOTE: How much time to wait before shifting the table window by 'TABLE_SHIFT_COUNT' tables.
+const TABLE_SHIFT_INTERVAL_S = latte::param!("table_shift_interval_s", 10);
```
I think those needs a bit more explaining

fruch commented on data_dir/latte/schema_scale.rn:
```diff
@@ -0,0 +1,277 @@
+        }
+        let data = gen_data(idx, ks_index, t_index).await;
+        for col_name in data.col_names {
+            // TODO: make it be compatible with prep stmts which select non-all columns
```
raise issue for it, as followup for this one

vponomaryov replied:
Moved the rune script into its own PR:
- https://github.com/scylladb/scylla-cluster-tests/pull/12954

Also, the rune script is not merged, so, there is no required condition to file a QA task

vponomaryov commented:
@fruch changes since our last talk about it:
- Split the rune script into its own PR: https://github.com/scylladb/scylla-cluster-tests/pull/12954
- Added second test case - for the vnodes
- Moved the jenkins files out of the `tier2` dir and placed in the `scale` one
- Removed `debug` stuff from the test configs.
- Added some more comments with useful info
- Enabled the nemesis for the `mixed` phase.

Summary:
- Tablets test case is going to fail on the `data population phase` because of the bugs:
  - https://github.com/scylladb/scylladb/issues/27620
  - https://github.com/scylladb/scylladb/issues/27621
- Vnodes test case must work
  - The https://github.com/scylladb/scylladb/issues/27566 is workarounded by the smaller latte pressure during the `data population` phase
  - The `mixed` phase was not tested with nemesis, but since it is the last phase in the test, consider it a ready to go state

pehala commented:
> * It is directly related to the cache hit/miss control - we will have a relatively stable load with a relatively stable (controlled) cache hit/miss ratio using all of the tables smoothly switching/shifting among them.

This is very useful, but I do not know how to use this, I think we should enhance this in the future with a clear CLI option, like `cache_hit_rate: 0.8` so we more people could use it in their testing

vponomaryov commented:
Rebased to the latest master which has now the rune script used in these new config files.
